scripts/parse-criterion-output.awk

```diff
@@ -0,0 +1,114 @@
+###
+# This script matches criterion output to output JSON for use for formatting output in CI
+#
+# We'd prefer to use machine output from cargo-criterion but there is an
+# outstanding issue preventing this:
+# https://github.com/bheisler/cargo-criterion/issues/17
+#
+# It matches output like:
+# ```
+# Benchmarking transforms/transforms
+# Benchmarking transforms/transforms: Warming up for 3.0000 s
+# Benchmarking transforms/transforms: Collecting 100 samples in estimated 7.6949 s (200 iterations)
+# Benchmarking transforms/transforms: Analyzing
+# transforms/transforms   time:   [31.562 ms 31.932 ms 32.307 ms]
+#                         thrpt:  [32.471 MiB/s 32.852 MiB/s 33.238 MiB/s]
+#                  change:
+#                         time:   [-0.5510% +1.0918% +2.8006%] (p = 0.19 > 0.05)
```

Comment:
I note the `p` value here along with the `noise-threshold` setting can be used to have criterion detect and output these kind of markers within a specific statistical threshold: https://bheisler.github.io/criterion.rs/book/user_guide/command_line_output.html#change

![image](https://user-images.githubusercontent.com/130903/100921119-f7aa4c00-3490-11eb-94bc-67463b6af936.png)

Reply:
:+1: we are currently using the default p-value threshold of `0.05`. As I understand it, criterion will use this to determine whether to output that the performance changed (which is the most important bit).

Reply:
This is also a value we can tweak: https://docs.rs/criterion/0.3.3/criterion/struct.BenchmarkGroup.html#method.significance_level